website/blog/2022-12-12-benchmarking-intel-vs-arm-for-backup-applications.md

```diff
@@ -0,0 +1,237 @@
+| --- | --- | --- | --- | --- |
+| Intel | m6i.xlarge | 4 | 16 | $0.192 |
+| ARM | m6g.xlarge | 4 | 16 | $0.154 |
+
+## Experiment
+
+In Corso, we use [Kopia](https://kopia.io/) as our data processing
+engine. It first split the data to divide larger files into smaller
+chunks for efficient deduplication. That split data is then
+deduplicated through the use of cryptographically strong hashing
+algorithms, compressed, and then encrypted.
+
+Based on this data pipeline, we split our measurements into 5
+different parts to microbenchmark the different parts of the above
+process. For each of the below microbenchmarks, we also varied
+internal benchmark parallelism between 1, 4, and 8 parallel
+goroutines.
+
+- Splitting: Buzhash vs. Rabin-Karp algorithms
+- Compression: Different variants of deflate, S2, lz4, zstd, gzip, and pgzip
+- Hashing: Different BLAKE and HMAC-SHA variants
+- Encryption: AES256-GCM and ChaCha20-Poly1305
+- Hashing + Encryption: Combination of the above Encryption and Hashing algorithms
+
+## ARM vs. Intel performance results
+
+Given the large number of results collected (there are 372 distinct
+performance data points!), we will only be presenting a subset of the
+results collected for the splitting, compression, and combined hashing
+and encryption microbenchmarks.
+
+Within these results, we will focus on the numbers with parallelism
+set to 8 and will focus on throughput numbers (higher is better)
+measured in MiB/s. For easier visualization, we also present the
+relative performance where we compare the throughput of ARM
+vs. Intel. For a more price-accurate comparison, we also present the
+price-adjusted relative performance where we see how many more MiB/s
+we can process for the same dollar.
+
+While there is color on some of the results not presented but give us
+a shout on [Discord](https://discord.gg/63DTTSnuhT) if you want to see
+the raw data.
+
+### Splitter
+
+Let’s first look at splitting, a critical part of data deduplication,
+that uses rolling-hash functions. While there is a lot more
+information available elsewhere, a rolling hash is a type of hash
+function that operates on a sliding window of data, allowing it to
+efficiently compute the hash value for a given chunk of data without
+having to process the entire dataset. This is useful in deduplication
+because it allows Corso and Kopia to quickly compare the hash values
+of different chunks of identical data, even if they're not the same
+size or don't start at the same point in the dataset.
+
+The size of the blocks and the algorithm used to divide the dataset
+are important considerations in deduplication, as they can affect the
+speed and efficiency of deduplication. For these results we compare
+the performance of Buzhash vs. the Rabin-Karp algorithm with block
+sizes of 1MiB, 2MiB. 4MiB, and 8MiB.
+
+![Throughput of Corso splitters](./images/benchmarking/splitter_througput.png)
```

Comment:
<img width="759" alt="Screen Shot 2022-12-09 at 8 29 27 AM" src="https://user-images.githubusercontent.com/585019/206748039-dec890b2-ba61-457e-bfe0-5438f91da63c.png">
We either need to style these with a background or add some kind of bg to the .png, this isn't super readable